### ISSUE TITLE
Iterators.Cycle should not implement reverse and last

### ISSUE THREAD

**Issue by knuesel** [JuliaLang/julia#53017]
`Iterators.Cycle` implements `reverse` by reversing the underlying iterator. That's not the semantics of `reverse`, it may be a useful operation but `reverse` is supposed to give the iterator sequence in reverse. It seems to me that `reverse` and `last` should be undefined for infinite iterators such as `Cycle`.

Since the current behavior doesn't match the documentation, I guess it can be considered a bug an be fixed?

**Comment by adienes** [comment]
this one I would not call a "correctness bug," rather just an opportunistic hijacking/punning of some functions by `Cycle` (although I do agree with the premise of the issue)

**Comment by vtjnash** [comment]
Is the issue that, in a logical sense, at the infinite limit, it is not correct to say (in a logic proof) that the infinite sequence ends with the last element of the underlying cycle? I am not sure that is entirely relevant here, since `Cycle` isn't (hopefully) being used in a logic proof to disprove infinity.

**Comment by adienes** [comment]
> it is not correct to say (in a logic proof) that the infinite sequence ends with the last element of the underlying cycle?

I would not consider this correct

**Comment by vtjnash** [comment]
Right, it is not a property of an infinite set, but then what makes this a correctness bug?

**Comment by adienes** [comment]
I would also not consider this a correctness bug :) 

but I do think it is a little surprising of a design choice, especially for `last`
